File: 1512_good_pairs/solution.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        
        std::map<int, int> pairs;
        int count =0;

        for (int i=0; i< nums.size(); i++) {
            if(pairs.find(nums[i]) != pairs.end()) pairs[nums[i]]++;
            else{
                pairs[nums[i]] = 1;
            }
        }
        // If a number occurs n times, there are n*(n-1)//2 viable piars
        for (auto el: pairs){
            count+= (el.second * (el.second - 1)) / 2;
        }

        return count;
    }
};

File: 1512_good_pairs/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int numIdenticalPairs(int[] nums) {

        int goodPairs = 0;
        Map<Integer, Integer> freq = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            if (freq.containsKey(nums[i])) {
                goodPairs = goodPairs + freq.get(nums[i]);
                freq.put(nums[i], freq.get(nums[i])+1);
            }
            else {
                freq.put(nums[i], 1);
            }
        }
        return goodPairs;
    }

    public static void main(String args[]){
        return;
    }
}